### PR TITLE
tests(helpers): add a reopen option to proxy http clients

### PR DESCRIPTION
**Pull request by flrgh**
This takes the re-open logic I used in #8925 and moves it into the helper http client so it can be reused elsewhere.

While there are a lot of tests in kong that could benefit from this behavior, there's not really a huge benefit to updating existing, non-flaky tests that simply create a new http client for each test case, so for now I haven't changed any test cases aside from the obvious prometheus one.

### CODE REVIEW

**Comment by Tieske** [comment]
This is a useful change to the http-client. But exactly because of that, I think it should be an upstream-PR.
A related one is open on the upstream repo, to not re-use a client if the body hasn't been read.

**Review by jschmid1 — APPROVED** [review]
lgtm!

**Comment by flrgh** [comment]
Going to wait until #8933 is merged and rebase because it should make this test less flaky:

>  @github-actions Build & Test / Postgres integration - first (01-04) tests (pull_request) 

```
         FAILED  spec/02-integration/02-cmd/12-hybrid_spec.lua:141: kong hybrid with #postgres backend quits gracefully
spec/02-integration/02-cmd/12-hybrid_spec.lua:150: Expected objects to be equal.
Passed in:
(string) 'cat: /home/runner/work/kong/kong/servroot2/pids/nginx.pid: No such file or directory
'
Expected:
(string) ''

stack traceback:
	spec/02-integration/02-cmd/12-hybrid_spec.lua:150: in function <spec/02-integration/02-cmd/12-hybrid_spec.lua:141>
```

**Pull request merged**
